broker/lib/controllers/ServiceBrokerApiController.js

```diff
@@ -105,22 +173,83 @@ class ServiceBrokerApiController extends FabrikBaseController {
       .then(done);
   }
 
+  patchInstanceV2(req, res) {
+    const params = _
+      .chain(req.body)
+      .omit('plan_id', 'service_id')
+      .cloneDeep()
+      .value();
+    //cloning here so that the DirectorInstance.update does not unset the 'service-fabrik-operation' from original req.body object
+
+    function done(result) {
+      let statusCode = CONST.HTTP_STATUS_CODE.OK;
+      const body = {};
+      if (req.instance.async) {
+        statusCode = CONST.HTTP_STATUS_CODE.ACCEPTED;
+        body.operation = utils.encodeBase64(result);
+      } else if (result && result.description) {
+        body.description = result.description;
+      }
+      res.status(statusCode).send(body);
+    }
+
+    req.operation_type = CONST.OPERATION_TYPE.UPDATE;
+    this.validateRequest(req, res);
+
+    return Promise
+      .try(() => {
+        if (!req.manager.isUpdatePossible(params.previous_values.plan_id)) {
+          throw new BadRequest(`Update to plan '${req.manager.plan.name}' is not possible`);
+        }
+        return Promise.try(() => {
+            if (req.manager.name === CONST.INSTANCE_TYPE.DIRECTOR) {
+              // Acquire lock for this instance
+              return lockManager.lock(req.params.instance_id, {
+                lockType: CONST.ETCD.LOCK_TYPE.WRITE,
+                lockedResourceDetails: {
+                  resourceType: CONST.APISERVER.RESOURCE_TYPES.DEPLOYMENT,
+                  resourceName: CONST.APISERVER.RESOURCE_NAMES.DIRECTOR,
+                  resourceId: req.params.instance_id,
+                  operation: CONST.OPERATION_TYPE.UPDATE
+                }
+              });
+            }
+          })
+          .then(() => req.instance.update(params));
+      })
+      .then(done)
+      .catch(err => {
+        if (err instanceof EtcdLockError) {
+          throw err;
+        }
+        return lockManager.unlock(req.params.instance_id)
```

Comment:
There might be a case where unlock is triggered even though lock was not taken. Solve it using the same as startBackup.

Reply:
Done

Reply:
As of now no TTL on locks (That means infinite TTL). Thinking of keeping TTL to 1 day? Need to come up with optimal TTL.
We are retrying unlock 3 [times](https://github.com/cloudfoundry-incubator/service-fabrik-broker/pull/304/files#diff-ed21a3386305ab1aa4301c46d87b0e98R154).